app/src/main/java/com/gmail/nowak/wjw/popularmovies/data/repository/MoviesRepository.java

```diff
@@ -0,0 +1,324 @@
+    private MutableLiveData<ApiResponseMovieListObject> topRatedMoviesResponseLD = new MutableLiveData<>();
+    private MutableLiveData<ApiResponseMovieListObject> popularMovieResponseLD = new MutableLiveData<>();
+    private LiveData<List<FavouriteMovie>> favMoviesData;// = new MutableLiveData<>();
+    private MutableLiveData<ApiMovie> apiMovieDetails = new MutableLiveData<>();
+    public AppDatabase database;
+    OkHttpClient okHttpClient;
+
+    public static MoviesRepository getInstance(Application application) {
+        if (sInstance == null) {
+            synchronized (LOCK) {
+                Timber.d("Creating new repository instance");
+                sInstance = new MoviesRepository(application);
+            }
+        }
+        Timber.d("Getting the repository instance");
+        return sInstance;
+    }
+
+    private MoviesRepository(Application application) {
+        database = AppDatabase.getInstance(application);
+
+        okHttpClient = setUpOkHttpClient();
+
+        retrofit = new Retrofit.Builder().baseUrl(NetworkUtils.THE_MOVIE_DATABASE_API_BASE_URL)
+                .client(okHttpClient)
+                .addConverterFactory(GsonConverterFactory.create())
+                .build();
+        theMovieDatabaseOrgAPI = retrofit.create(TheMovieDataBaseOrgAPI.class);
+    }
+
+    //todo Q? move setUpOkHttpClient and all this retrofit preparing from constructor to another class? ie NetworkClient.getTheMovieDatabaseClient()?
+
+    /**
+     * Set up HttpClient intercepting with api key param
+     *
+     * @return
+     */
+    private OkHttpClient setUpOkHttpClient() {
+        OkHttpClient.Builder httpClientBuilder =
+                new OkHttpClient.Builder();
+        httpClientBuilder.addInterceptor(new Interceptor() {
+            @Override
+            public Response intercept(Chain chain) throws IOException {
+                Request original = chain.request();
+                HttpUrl originalHttpUrl = original.url();
+
+                HttpUrl newUrl = originalHttpUrl.newBuilder()
+                        .addQueryParameter(NetworkUtils.API_KEY_PARAM, BuildConfig.TMD_API_KEY)
+                        .build();
+
+                // Request customization: add request headers
+                Request.Builder requestBuilder = original.newBuilder()
+                        .url(newUrl);
+
+                Request newRequest = requestBuilder.build();
+                return chain.proceed(newRequest);
+            }
+        });
+        return httpClientBuilder.build();
+    }
+
+    /**
+     * Fetching data - list of movies from theMovieDatabase.org
+     *
+     * @param category either topRated or popular
+     * @param page     unused, in the future implementing the possibility of fetching more than just one page
+     */
+    private void fetchMovieListFromApi(final String category, Integer page) {
+        //cancel all previous calls
+        okHttpClient.dispatcher().cancelAll();
```

Comment:
This doesn't look good. With this, you can cancel also all other calls that might be running in the background. I understand that this is small app - but I don't think it's a good approach.
Better would be to:
1) prevent multiple calls at the same time, or
2) allowing past calls to resolve (and handle response properly), or
3) if needed cancel only necessary calls